### DISCUSSION TITLE
CAN Database Addition

### DISCUSSION THREAD

**Discussion by tbray21** [cantools/cantools#420] [General]
I am using the CAN/DBC database code to take apart the dbc files. I need to take a set of dbc files across several CAN busses and then organize them by signal, not messages, to look for inconsistencies in value format and enumerated values. I would like to add a property to the signal class that will allow pulling a copy of the choices dictionary, possibly getting it as either a dictionary or a list of tuples.

I currently just access the self._choices directly but that violates my coding style (being an old C/C++ developer, I like private variables). My idea is to add a property to the signal class that would return a copy of _choices

Any comments or suggestions?

Tom Bray

**Comment by andlaus** [comment]
doesn't
```python
from copy import deepcopy
# [...]
signal_choice_copy = deepcopy(signal.choices)
```
work for you? (this is basically python's standard pattern in to do deep copies of objects)

**Reply by tbray21** [reply]
Yes, I just went back through the properties and found it. I don't know how I missed it.
